rllib/policy/view_requirement.py

```diff
@@ -28,90 +31,91 @@ class ViewRequirement:
         >>> req = ModelV2(...).view_requirements # doctest: +SKIP
         >>> print(req) # doctest: +SKIP
         {"obs": ViewRequirement(shift=0)}
+
+    Args:
+        data_col: The data column name from the SampleBatch
+            (str key). If None, use the dict key under which this
+            ViewRequirement resides.
+        space: The gym Space used in case we need to pad data
+            in inaccessible areas of the trajectory (t<0 or t>H).
+            Default: Simple box space, e.g. rewards.
+        shift: Single shift value or
+            list of relative positions to use (relative to the underlying
+            `data_col`).
+            Example: For a view column "prev_actions", you can set
+            `data_col="actions"` and `shift=-1`.
+            Example: For a view column "obs" in an Atari framestacking
+            fashion, you can set `data_col="obs"` and
+            `shift=[-3, -2, -1, 0]`.
+            Example: For the obs input to an attention net, you can specify
+            a range via a str: `shift="-100:0"`, which will pass in
+            the past 100 observations plus the current one.
+        index: An optional absolute position arg,
+            used e.g. for the location of a requested inference dict within
+            the trajectory. Negative values refer to counting from the end
+            of a trajectory. (#TODO: Is this still used?)
+        used_for_compute_actions: Whether the data will be used for
+            creating input_dicts for `Policy.compute_actions()` calls (or
+            `Policy.compute_actions_from_input_dict()`).
+        used_for_training: Whether the data will be used for
+            training. If False, the column will not be copied into the
+            final train batch.
     """
 
-    def __init__(
-        self,
-        data_col: Optional[str] = None,
-        space: gym.Space = None,
-        shift: Union[int, str, List[int]] = 0,
-        index: Optional[int] = None,
-        batch_repeat_value: int = 1,
-        used_for_compute_actions: bool = True,
-        used_for_training: bool = True,
-    ):
+    data_col: Optional[str] = None
+    space: gym.Space = None
+    shift: Union[int, str, List[int]] = 0
+    index: Optional[int] = None
+    batch_repeat_value: int = 1
+    used_for_compute_actions: bool = True
+    used_for_training: bool = True
+    shift_arr: Optional[np.ndarray] = dataclasses.field(init=False)
+
+    def __post_init__(self):
```

Comment:
sadly, I introduced a couple of serialization functions to_dict() and from_dict().
do you know if this __post_init__ will get executed as well the way I construct things?

Reply:
yes. because in from_dict(0 you call data classes' constructor which calls post_init() at the end.